paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PrimaryButtonNew.kt

```diff
@@ -0,0 +1,370 @@
+        } else {
+            Color(style.getBackgroundColor(context))
+        }
+    }
+
+    val animatedBackground = if (LocalInspectionMode.current) {
+        background
+    } else {
+        animateColorAsState(
+            targetValue = background,
+            label = "BackgroundAnimation",
+            animationSpec = tween(durationMillis = 100, easing = LinearEasing)
+        ).value
+    }
+
+    val borderStroke = BorderStroke(style.shape.borderStrokeWidth.dp, Color(style.getBorderStrokeColor(context)))
+    val shape = RoundedCornerShape(style.shape.cornerRadius.dp)
+
+    CompositionLocalProvider(
+        LocalContentAlpha provides if (enabled) ContentAlpha.high else ContentAlpha.disabled,
+    ) {
+        Box(
+            modifier = Modifier.fillMaxWidth(),
+            contentAlignment = Alignment.Center,
+        ) {
+            TextButton(
+                onClick = onClick,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .defaultMinSize(
+                        minHeight = dimensionResource(id = R.dimen.stripe_paymentsheet_primary_button_height)
+                    ),
+                enabled = enabled,
+                shape = shape,
+                border = borderStroke,
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = animatedBackground,
+                    disabledBackgroundColor = animatedBackground,
+                ),
+            ) {
+                Content(
+                    label = label,
+                    processingState = processingState,
+                    locked = locked,
+                    onProcessingCompleted = onProcessingCompleted,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun Content(
+    label: String,
+    processingState: PrimaryButtonProcessingState,
+    locked: Boolean,
+    onProcessingCompleted: () -> Unit,
+) {
+    AnimatedContent(
+        processingState is PrimaryButtonProcessingState.Completed,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(IntrinsicSize.Max),
+        transitionSpec = { fadeAnimation },
+        label = "ContentAnimation",
+    ) { completed ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(horizontal = 8.dp)
+        ) {
+            if (completed) {
+                Complete(onProcessingCompleted)
+            } else {
+                Incomplete(
+                    text = when (processingState) {
+                        is PrimaryButtonProcessingState.Idle -> label
+                        else -> stringResource(R.string.stripe_paymentsheet_primary_button_processing)
+                    },
+                    processing = processingState !is PrimaryButtonProcessingState.Idle,
+                    locked = locked,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun BoxScope.Incomplete(
```

Comment:
I think we should rename this. It's a bit confusing as to what this does until you read the implementation.